app/src/main/java/org/thoughtcrime/securesms/conversation/v2/messages/VisibleMessageView.kt

```diff
@@ -303,8 +302,7 @@ class VisibleMessageView : LinearLayout {
 
         // --- If we got here then we know the message is outgoing ---
 
-        val thisUsersSessionId = TextSecurePreferences.getLocalNumber(context)
-        val lastSentMessageId = mmsSmsDb.getLastSentMessageFromSender(message.threadId, thisUsersSessionId)
+        val lastSentMessageId = mmsSmsDb.getLastOutgoingMessage(message.threadId)
```

Comment:
Yup, it gets called for every outgoing message - and it has to look at all the messages, ignoring the incoming ones that we don't show delivery status on, and taking OFF the delivery status on any previously sent messages which have successfully sent (but not any which have failed to send) and then turning ON visibility for the new outgoing message.

Then when other messages might be incoming it needs to do that check again in case the new message was outgoing from us. It's not very efficient, but without modifying the database schema so that we have a last outgoing message Id so we don't have to calculate it I don't know how to do it efficiently. You, or Morgan, or Fanchao might - but I've tried a dozen different ways to about it and I can't seem to find a nice way to do it.

As such, rather than spin my wheels fruitlessly on this I can raise a ticket for further refinement, and this PR will basically prevent the constant respawn of attachment download jobs w/ some minor cleanup of the outgoing message check code.

Reply:
this value was supplied in onBind a second ago? can we just use that?

Reply:
This lookup could be causing [this](https://github.com/oxen-io/session-android/issues/1450)

I don't think my suggestion nor this implementation is perfect, as when we get a new message, that could invalidate the state of a past message, but worst case scenario is we show a status when we shouldn't. But right now, the app can become so slow it's unusable. So for that reason I want to remove the `O( n * m )` lookups from the main thread at the least. We can remove all db lookups on main later. But we can't go searching for things on main, let alone more times than is strictly necessary, while scrolling, as `onBind` is typically called in response to scrolling.

If (in the future) we want to make sure the last item is definitely the only one that's forced to be shown then we either

1. Make each item listen for changes to the lastSentMessage, and post there once, with one db lookup... or
2. Call notifyItemsChanged or similar and rebind the views or update them with this payload.

but we can't search the db here in `onBind`.

Reply:
Have adjusted to update the last sent message in create/resume and in the `onChanged` method of the recycler data observer.